src/main/java/org/candlepin/resource/ProductResource.java

```diff
@@ -120,13 +120,14 @@ public ProductCertificateDTO getProductCertificate(String productUuid) {
     }
 
     @Override
-    public CandlepinQuery<OwnerDTO> getProductOwners(List<String> productUuids) {
+    public Iterable<OwnerDTO> getProductOwners(List<String> productUuids) {
         if (productUuids.isEmpty()) {
             throw new BadRequestException(i18n.tr("No product IDs specified"));
         }
 
-        return this.translator.translateQuery(
-            this.ownerCurator.getOwnersWithProducts(productUuids), OwnerDTO.class);
+        List<Owner> owners = this.ownerCurator.getOwnersWithProducts(productUuids);
+        return owners.stream()
+            .map(this.translator.getStreamMapper(Owner.class, OwnerDTO.class)).collect(Collectors.toList());
```

Comment:
We don't need to collect this into a list -- we can output a stream of DTOs.

Reply:
Changed it to Iterable, along with new Pagination logic.